test-network-function/generic/suite.go

```diff
@@ -885,6 +896,53 @@ func uncordonNode(node string) {
 	gomega.Expect(testResult).To(gomega.Equal(tnf.SUCCESS))
 }
 
+// Pod antiaffinity test for all deployments
+func testPodAntiAffinity(podNamespace string) {
+	var deployments dp.DeploymentMap
+	ginkgo.When("CNF is designed in high availability mode ", func() {
+		ginkgo.It("Should set pod replica number greater than 1 and corresponding pod anti-affinity rules in deployment", func() {
+			defer results.RecordResult(identifiers.TestPodHighAvailabilityBestPractices)
+			deployments, _ = getDeployments(podNamespace)
+			if len(deployments) == 0 {
+				return
+			}
+			for name, d := range deployments {
+				if name != partnerPod {
+					podAntiAffinity(name, podNamespace, d.Replicas)
+				}
+			}
+		})
+	})
+}
+
+// check pod antiaffinity definition for a deployment
+func podAntiAffinity(deployment, podNamespace string, replica int) {
+	context := getContext()
+	values := make(map[string]interface{})
+	values["DEPLOYMENT_NAME"] = deployment
+	values["DEPLOYMENT_NAMESPACE"] = podNamespace
+	infoWriter := tnf.CreateTestExtraInfoWriter()
+	test, handlers, result, err := generic.NewGenericFromMap(relativePodTestPath, relativeSchemaPath, values)
+	gomega.Expect(err).To(gomega.BeNil())
+	gomega.Expect(result).ToNot(gomega.BeNil())
+	gomega.Expect(result.Valid()).To(gomega.BeTrue())
+	gomega.Expect(handlers).ToNot(gomega.BeNil())
+	gomega.Expect(len(handlers)).To(gomega.Equal(1))
+	gomega.Expect(test).ToNot(gomega.BeNil())
+	tester, err := tnf.NewTest(context.GetExpecter(), *test, handlers, context.GetErrorChannel())
+	gomega.Expect(err).To(gomega.BeNil())
+	gomega.Expect(tester).ToNot(gomega.BeNil())
+
+	testResult, err := tester.Run()
+	if testResult != tnf.SUCCESS {
+		msg := fmt.Sprintf("The POD replica is found %d, and podAntiAffinity rule is not defined, you might want to change it in deployment %s in namespace %s",
```

Comment:
and podAntiAffinity rule is not defined, 
->
but podAntiAffinity rule is not defined or replica count is one,

Reply:
Thanks for the clarification Ying, please disregard previous comment. What about the following change:

if testResult != tnf.SUCCESS {
     if replica >1 {
		msg := fmt.Sprintf("The deployment replica count is %d, but a podAntiAffinity rule is not defined, you might want to change it in deployment %s in namespace %s",
replica, deployment, podNamespace)
} else {
                msg := fmt.Sprintf("The deployment replica count is %d. Pod replica should be >2 with an podAntiAffinity rule defined . You might want to change it in deployment %s in namespace %s",
replica, deployment, podNamespace)
}

Reply:
looks better, will make change